sdk/dapp-kit/src/actions/connectWallet.ts

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import type {
+	StandardConnectInput,
+	WalletAccount,
+	WalletWithSuiFeatures,
+} from '@mysten/wallet-standard';
+import type { Dispatch } from 'react';
+import type { WalletAction } from '../reducers/walletReducer.js';
+import type { StorageAdapter } from '../utils/storageAdapters.js';
+import { setMostRecentWalletConnectionInfo } from '../utils/walletUtils.js';
+
+export type ConnectWalletArgs = {
+	/** The wallet to connect to. */
+	wallet: WalletWithSuiFeatures;
+
+	/** An optional account address to connect to. Defaults to the first authorized account. */
+	accountAddress?: string;
+} & StandardConnectInput;
+
+export async function connectWallet(
+	dispatch: Dispatch<WalletAction>,
+	storageAdapter: StorageAdapter,
+	storageKey: string,
+	{ wallet, accountAddress, ...standardConnectInput }: ConnectWalletArgs,
+) {
+	dispatch({ type: 'wallet-connection-status-updated', payload: 'connecting' });
+
+	try {
+		const connectResult = await wallet.features['standard:connect'].connect(standardConnectInput);
+		const selectedAccount = getSelectedAccount(connectResult.accounts, accountAddress);
+
+		dispatch({
+			type: 'wallet-connected',
+			payload: { wallet, currentAccount: selectedAccount },
+		});
+
+		await setMostRecentWalletConnectionInfo({
```

Comment:
Why not just make this an effect in the autoconnect logic? Feels wrong to me to have this here.

Reply:
Plus then you can colocate the getting from storage + the persisting to storage in a single place

Reply:
Just persist that info after state changes in an effect?

Reply:
OH sorry, I see what you're saying. Hmm, my thinking was that storage changes based on events (disconnect, connect, switching accounts, etc.) more so than changing as a side effect of currentWallet / currentAccount changing.

I think applying that principle beyond connection makes the logic a little more complex as well, because now we would need to track the previously connected wallet / account in state to determine whether or not to remove info from storage or update storage.

Reply:
Yeah there is some subtle nuance between events-based and effect based. From what I can tell, disconnection is the only thing that gets a little weird, but seems pretty solvable.

Here's roughly what I'm imagining this would look like all together:
```ts
function useAutoConnect(autoConnect: boolean) {
  const { wallet, account } = useWalletContext();
  const previousWallet = usePrevious(wallet);
  const previousAccount = usePrevious(account);
  useEffect(() => {
    if (!autoConnect) return;

    // NOTE: We use the previous wallet comparison to avoid immediately removing the storage value on initial mount
    if (wallet !== previousWallet || account !== previousAccount) {
      if (wallet) {
        // connect:
        storage.set({ wallet, account });
      } else {
        // disconnect:
        storage.delete();
      }
    }
  }, [wallet, account, previousWallet, previousAccount]);

  useEffect(() => {
    if (autoConnect) {
      const { wallet, account } = storage.get();
      connect(wallet, account);
    }
  }, [autoConnect]);
}
```

In effect it's pretty similar, but I think the added benefit is that all of the autoconnection logic is entirely separate and not part of the core connection flow. This isn't really "essential" by any means, to my brain it's just a little "cleaner" (I also think it makes things like testing more easy, because things are more decoupled).

If you don't want to go down this route that's also fine, I'm not dead set on this, more I just imagined this being the rough way we'd do this to begin with.